UsageControlFramework/src/main/java/it/cnr/iit/usagecontrolframework/proxies/ProxyPDP.java

```diff
@@ -74,40 +72,42 @@ private boolean buildLocalPdp( PdpProperties properties ) {
 
     @Override
     public PDPEvaluation evaluate( String request, String policy ) {
-        if( initialized ) {
-            return pdp.evaluate( request, policy );
+        switch( getConnection() ) {
+            case API:
+                Reject.ifNull( pdp );
+                return pdp.evaluate( request, policy );
+            default:
+                return null;
         }
-        return null;
     }
 
     @Override
-    public PDPEvaluation evaluate( String request ) {
-        return null;
-    }
-
-    public void setInterfaces( ProxyPAP proxyPAP ) {
-        Reject.ifNull( proxyPAP );
+    public PDPEvaluation evaluate( String request, PolicyWrapper policyHelper, STATUS status ) {
+        switch( getConnection() ) {
+            case API:
+                Reject.ifNull( pdp );
 
-        if( getConnection() == CONNECTION.API
-                && pdp.isInitialized() ) {
-            pdp.setPAPInterface( proxyPAP );
-            initialized = true;
+                return pdp.evaluate( request, policyHelper, status );
+            default:
+                return null;
         }
     }
 
     @Override
-    public PDPEvaluation evaluate( String request, StringBuilder policy, STATUS status ) {
-        if( pdp != null ) {
-            return pdp.evaluate( request, policy, status );
-        }
+    public PDPEvaluation evaluate( String request ) {
         return null;
     }
 
-    protected CONNECTION getConnection() {
-        return CONNECTION.valueOf( properties.getCommunicationType() );
+    public void setInterfaces( PAPInterface pap ) {
+        Reject.ifNull( pap );
+        pdp.setPAP( pap );
     }
 
     public boolean isInitialized() {
-        return initialized;
+        return pdp != null ? pdp.isInitialized() : false;
+    }
+
+    protected CONNECTION getConnection() {
+        return CONNECTION.valueOf( properties.getCommunicationType() );
     }
 }
```

Comment:
Returning null in multiple places in this class, can the caller handle null?

Reply:
ok now CH checks if the evaluation is null.